src/sentry/integrations/tasks/sync_status_outbound.py

```diff
@@ -66,26 +67,28 @@ def sync_status_outbound(group_id: int, external_issue_id: int) -> bool | None:
         action_type=ProjectManagementActionType.OUTBOUND_STATUS_SYNC, integration=integration
     ).capture() as lifecycle:
         lifecycle.add_extra("sync_task", "sync_status_outbound")
-        if installation.should_sync("outbound_status"):
-            lifecycle.add_extras(
-                {
-                    "organization_id": external_issue.organization_id,
-                    "integration_id": integration.id,
-                    "external_issue": external_issue_id,
-                    "status": group.status,
-                }
-            )
-            try:
+        try:
+            if installation.should_sync("outbound_status"):
+                lifecycle.add_extras(
+                    {
+                        "organization_id": external_issue.organization_id,
+                        "integration_id": integration.id,
+                        "external_issue": external_issue_id,
+                        "status": group.status,
+                    }
+                )
+
                 installation.sync_status_outbound(
                     external_issue, group.status == GroupStatus.RESOLVED, group.project_id
                 )
-            except (IntegrationFormError, ApiUnauthorized) as e:
-                lifecycle.record_halt(halt_reason=e)
-                return None
-            analytics.record(
-                "integration.issue.status.synced",
-                provider=integration.provider,
-                id=integration.id,
-                organization_id=external_issue.organization_id,
-            )
+
+                analytics.record(
+                    "integration.issue.status.synced",
+                    provider=integration.provider,
+                    id=integration.id,
+                    organization_id=external_issue.organization_id,
+                )
+        except (IntegrationFormError, ApiUnauthorized, OrganizationIntegrationNotFound) as e:
```

Comment:
I know this is old code, but will this ever yield an IntegrationFormError? Theoretically it shouldn't